src/dawn/Optimizer/PassSetNonTempCaches.cpp

```diff
@@ -44,12 +44,20 @@ struct NameToImprovementMetric {
 
 /// @brief The GlobalFieldCacher class handles the caching for a given Multistage
 class GlobalFieldCacher {
+  const std::unique_ptr<iir::MultiStage>& multiStagePrt_;
```

Comment:
I think the probem is that we have no convention. (Compare to StencilMetainformation). We should figure out what we want and write this down in the dev-guide. But this can be for later. I'll create a task doing both this and the refactoring